perfmetrics/scripts/bigquery/bigquery.py

```diff
@@ -0,0 +1,246 @@
+# Copyright 2023 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Python script for setting up tables in BigQuery and exporting performance metrics to them.
+
+This python script sets up creates tables in BigQuery and exports fio, vm as well as list
+metrics to them. It is also used ot insert new experiment configurations into the BigQuery tables
+or update them.
+
+Typical usage from bigquery folder:
+  $ python3 bigquery.py [-h] --gcsfuse_flags GCSFUSE_FLAGS --branch BRANCH --end_date END_DATE
+
+  Flag gcsfuse_flags (required): GCSFuse flags for mounting the test buckets.
+  Flag branch (required): GCSFuse repo branch to be used for building GCSFuse.
+  Flag end_date (required): Date upto when tests are run.
+
+Note:
+  Make sure BigQuery API is enabled for the project
+
+"""
+import argparse
+from google.cloud import bigquery
+import sys
+
+PROJECT_ID = 'gcs-fuse-test'
+DATASET_ID = 'performance_metrics'
+CONFIGURATION_TABLE_ID = 'experiment_configuration'
+FIO_TABLE_ID = 'fio_metrics'
```

Comment:
let's call the tables as read_write_fio_metrics, read_write_vm_metrics & list_metrics

Reply:
Okay!